### PR TITLE
Add test for hardcoded IPs in the K8s resource configuration cncf/cnf-conformance#68

### PR DESCRIPTION
**Pull request by denverwilliams**
# CNF Conformance PR Template

## Description
This PR adds the 'hardcoded_ip_addresses_in_k8s_runtime_configuration' conformance test.

## Issues:
Refs: cncf/cnf-conformance#68

## How has this been tested:
 - [ ] Covered by existing integration testing
 - [x] Added integration testing to cover
 - [x] Verified all A/C passes
     * [ ] develop
     * [ ] master
     * [x] tag/other branch
 - [x] Test environment
    * [x] Shared Packet K8s cluster
    * [ ] New Packet K8s cluster
    * [x] Kind cluster
 - [ ] Have not tested

## Types of changes:
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [x] Documentation update

## Checklist:
**Documentation**
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] No updates required.

**Code Review**
- [ ] Does the test handle fatal exceptions, ie. rescue block

**Issue**
- [x] Tasks in issue are checked off

### CODE REVIEW

File: src/tasks/configuration_lifecycle.cr [hunk 1]
```diff
@@ -231,11 +232,11 @@ task "nodeport_not_used", ["retrieve_manifest"] do |_, args|
     service_type = service.get("spec").as_h["type"].as_s
     puts service_type if check_verbose(args)
     if service_type == "NodePort" 
-      puts "FAILURE: NodePort is being used".colorize(:red)
+      puts "✖️  FAILURE: NodePort is being used".colorize(:red)
```
Comment (williscool) [inline src/tasks/configuration_lifecycle.cr:235]: I think the emoji's shouldn't be hardcoded. they should live in a central file somewhere

File: src/tasks/configuration_lifecycle.cr [hunk 2]
```diff
@@ -231,11 +232,11 @@ task "nodeport_not_used", ["retrieve_manifest"] do |_, args|
...
       upsert_failed_task("nodeport_not_used")
     else
-      puts "PASSED: NodePort is not used"
-      upsert_passed_task("nodeport_not_used").colorize(:green)
+      puts "✔️  PASSED: NodePort is not used".colorize(:green)
```
Comment (williscool) [inline src/tasks/configuration_lifecycle.cr:238]: same thing about hardcoded emojis

File: src/tasks/configuration_lifecycle.cr [hunk 3]
```diff
@@ -245,3 +246,46 @@ task "nodeport_not_used", ["retrieve_manifest"] do |_, args|
...
+    else
+      helm_install = `#{helm} install generated #{destination_cnf_dir}/#{helm_directory} --dry-run --debug > #{destination_cnf_dir}/helm_chart.yml`
+      puts "helm_directory: #{helm_directory}" if check_verbose(args)
+    end
+ 
+    ip_search = File.read_lines("#{destination_cnf_dir}/helm_chart.yml").take_while{|x| x.match(/NOTES:/) == nil}.reduce([] of String){|acc, x| x.match(/([0-9]{1,3}[\.]){3}[0-9]{1,3}/) && x.match(/([0-9]{1,3}[\.]){3}[0-9]{1,3}/).try &.[0] != "0.0.0.0" ? acc << x : acc}
+    puts "IPs: #{ip_search}" if check_verbose(args)
+
+    if ip_search.empty? 
+      puts "✔️  PASSED: No hard-coded IP addresses found in the runtime K8s configuration".colorize(:green)
```
Comment (williscool) [inline src/tasks/configuration_lifecycle.cr:277]: hardcoded emoji

File: src/tasks/configuration_lifecycle.cr [hunk 4]
```diff
@@ -245,3 +246,46 @@ task "nodeport_not_used", ["retrieve_manifest"] do |_, args|
...
+      upsert_passed_task("hardcoded_ip_addresses_in_k8s_runtime_configuration")
+    else
+      puts "✖️  FAILURE: Hard-coded IP addresses found in the runtime K8s configuration".colorize(:red)
```
Comment (williscool) [inline src/tasks/configuration_lifecycle.cr:280]: hardcoded emoji

**Comment by taylor** [comment]
No emoji's for passing A/C.  Everything else looks good.  Going to merge.  New PR needed for emoji

#68 

cc: @denverwilliams @williscool @lixuna

File: src/tasks/configuration_lifecycle.cr (hunk 2 above)
Comment (taylor) [inline src/tasks/configuration_lifecycle.cr:238]: Missing emojis for the test

File: src/tasks/configuration_lifecycle.cr (hunk 4 above)
Comment (taylor) [reply src/tasks/configuration_lifecycle.cr:280]: Missing emojis for the test

File: src/tasks/configuration_lifecycle.cr (hunk 3 above)
Comment (taylor) [reply src/tasks/configuration_lifecycle.cr:277]: Missing emojis for the test

**Pull request merged**